Improvements and fixes

DoctorBearPhD opened the pull request:
I made some changes and fixes. Take a look! :nerd_face: 
Improvements include: 
* New SFV version compatibility fixes
* Allowing RTE to scan for large-file character scripts (like Zeku and G)
* Single merged .exe file for each program (no separate .dll's)
* Various quality-of-life improvements 
  - (like ToolTips on RTE, and pausing for MoveTool when no arguments are passed so you can read what it says)

DoctorBearPhD committed:
Combined commit includes BCM fix and my 'improvements' and fixes

DoctorBearPhD committed:
Previous commit didn't actually have all the updates

* From updates: Added pause, fixed pause, cleaned up code and presentation

DoctorBearPhD committed:
Transfer BCM Fix

DoctorBearPhD committed:
Transfer RTE MainForm improvements

* Gave more specific names to ambiguous variables
* Added some comments
* Used string interpolation instead of concatenation

DoctorBearPhD committed:
Transfer improvement: RTE in one exe

DoctorBearPhD committed:
Fix/add tooltips

DoctorBearPhD committed:
Transfer fix: RTE Zeku/G/large-file fix

* Allows the use of RTE with Zeku, G, and any other large-file characters (<4000kB).
* Previous limit was 2000kB. Probably don't need more than 3000kB, but this gives some wiggle room.

DoctorBearPhD commented:
Note about some of the commits: There was a weird end-of-line inconsistency (CLRF) thing that caused some of the diffs to show entire files being replaced, even if it was just a couple lines. :sweat:

DoctorBearPhD committed:
In BAC.Move - Renamed nondescriptive property names, still able to read in old names

* Successful test of name conversion!

lullius commented:
Thanks DoctorBearPhD!

All of these changes looks good to me. The code was quite messy in places, I see you even cleaned it up a bit.

"* Successful test of name conversion!" means that there are no breaking changes since last version, right? If so, that's excellent!

About the diffs, I can't actually see what you changed in the files that are "replaced". It would be better if you could fix that somehow. It's going to be hard to find out what needs to be looked at later if your changes introduces new bugs.

Does "Single merged .exe file for each program (no separate .dll's)" mean that there's no separate MoveLib.dll anymore? I don't know if others are using movelib in their projects. I think sfvsim (https://gitlab.com/loic.petit/sfvsim) used to use it, but Loïc made his own tool I think (https://gitlab.com/loic.petit/sfvfile)? Would this change make it harder for others to use movelib in their projects?


Over to something completely different:
I think MoveTool needs a new maintainer. I'm not really into SF5, never was actually. Especially after those responsible (Capcom? Valve?) dropped the ball on linux support even after it was officially announced (http://www.capcom-unity.com/combofiend/blog/2015/12/17/announcing-steam-os-support-for-sfv).
I don't use Windows anymore, only linux, so maintaining MoveTool would also be difficult for me. Testing it is more or less impossible for me as SF5 doesn't run in Wine. At least that was the case last time I checked.

I think I've seen your nick before. Are you by any chance working on Mysterious Mod? I think MM is great and also one of the biggest projects that use MoveTool. Handing the MoveTool project over to someone working on MM feels like a natural move to me.
Loïc (@WydD) also used MoveTool when he was making some of his tools I think. He also has a very detailed wiki about how all the values work (https://gitlab.com/loic.petit/sfvsim/wikis/home).

If either of you, DoctorBearPhD and Loïc, or both, want to take over for me, that would be great. In that case, please clone this repository somewhere and give me the link. I will put it at the top of the readme so that everyone who comes here knows MoveTool moved and to where.

WydD commented:
Yeah I made my own movetool for sfvsim but I'm still using a modified version of movetool for sfvdiff (using bson not json) and for exploratory digging. I think that most of the work on Movetool has been made lately by @DoctorBearPhD (I only made a small change when season 3 started). Maybe we could just use his fork and start from there ? I'm not a MM contributor but he is and I think that MM is the prime user of Movetool now. That said, @DoctorBearPhD, you know that if you need help you know where to find me :)

lullius commented:
That sounds good to me, @WydD.
Would that be okay with you @DoctorBearPhD?

DoctorBearPhD commented:
Correct; no breaking changes since last version.
I'll look up some solutions to the whole-file-replaced issue.
Correct, the single .exe means that users only need the .exe files to run the programs. I imagine if someone wants MoveLib.dll, they could get it from this repo? :man_shrugging: The visual studio output creates a .dll, when it builds MoveLib.sln so it could be added with release if needed.
Why yes, I am working on MM :smiley: and made a GUI tool (for Windows) for modding SFV. Still needs work, so it's not released outside of the Discord server.
Loïc's Wiki has saved me tons of hassle and taught me a bunch about the values. :pray:

Sure, I'd love to take over as maintainer of MoveTool. :+1:
It's been interesting, to say the least. :bear:

DoctorBearPhD commented:
In the diff viewer, it looks like you can click `Diff Settings` > `Hide whitespace changes`

lullius commented:
Thanks, that's good to hear!
I'll update the readme to point people who come here to your fork.

A GUI would be nice for sure, I know lots of people get scared by the json.

Getting the dll from this repository means people would get an outdated version after a while though. Anyone can just get the updated source and build it of course, so I guess it's fine.

You're right about the Diff Settings. I didn't know about that. I wouldn't worry about the "replaced" files then.

As I've said, I'm not really into SF5, but I do watch some MM-stuff once in a while. It looks great. It looks like more fun than the regular game to be honest.

Keep up the great work guys!

DoctorBearPhD committed:
More renames, comments, and cleanup

DoctorBearPhD committed:
Update README.md
